NeuroID/src/main/java/com/neuroid/tracker/service/NIDJobServiceManager.kt

```diff
@@ -40,16 +46,49 @@ object NIDJobServiceManager {
         jobCaptureEvents = createJobServer()
     }
 
+    /**
+     * Creates a job that synchronizes all requests to send events.
+     */
+    private fun createSendEventsServer(): Job {
+        return CoroutineScope(Dispatchers.IO).launch {
+            try {
+            for (notification in sendEventsNotification) {
+                var sendNow = notification
+                var next = sendEventsNotification.tryReceive()
+                while (next.isSuccess) {
+                    next.getOrNull()?.let {
+                        sendNow = sendNow || it
+                    }
+                    next = sendEventsNotification.tryReceive()
+                }
+                sendEventsNow(sendNow)
+            }
+            } catch (exception: Exception) {
+                NIDLog.e("NeuroID", exception.toString())
+            }
+            NIDLog.e("NeuroID", "send event job exited")
+        }
+    }
+
     private fun createJobServer(): Job {
         return CoroutineScope(Dispatchers.IO).launch {
             while (userActive && isActive) {
                 delay(5000L)
-                sendEventsNow()
+                sendEventsNotification.send(false)
             }
         }
     }
 
-    suspend fun sendEventsNow(forceSendEvents: Boolean = false) {
+    /**
+     * Notify the send events server to initiate a send.
+     */
+    fun sendEvents(forceSendEvents: Boolean = false) {
+        CoroutineScope(Dispatchers.IO).launch {
+            sendEventsNotification.send(forceSendEvents)
+        }
```

Comment:
Since `sendEventsNotification` has an unlimited buffer, we can drop the coroutine and use trySend:
https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.channels/-send-channel/#-1976436467%2FFunctions%2F1975948010